### ISSUE TITLE
Custom Component json to kamelet yaml

### ISSUE THREAD

**Issue by skumaravelan** [apache/camel-kamelets#1741]
For custom component services, we have written components.json file and and have corresponding classes in the camel runtime to execute the integrations . I am trying to figure out a mechanism of using the custom components.json file, i would like to define an kamelet.yaml using any pre existing tools by camel. so that we can use those tools to convert to kamelet file and update as needed using a Karavan kind of visual tool. I understand that these are two different ways of executing the components. As Karavan supports kamelet it works elegantly when we have these files converted using a camel cli or something along those line. Any suggestion on this front would help.

**Comment by oscerd** [comment]
Kamelets expects the components to be in the camel catalog. So it's not possible as of today to use directly a custom component through them.

**Comment by skumaravelan** [comment]
Is there a bootstrap utility  program to convert json to kanelet.

**Comment by oscerd** [comment]
No, there isn't. But if you're take into consideration custom component, even if you would be able to convert from json to Kamelets, the Kamelets wouldn't work.

**Comment by skumaravelan** [comment]
What are the consideration and requires to work?

**Comment by oscerd** [comment]
The requirements would be to support a custom catalog, but this is not related to Kamelets, it's more something more general @davsclaus

**Comment by oscerd** [comment]
Going to close this, since it's not on the roadmap and it's not related to this repository
